### PR TITLE
daphne: Add query type to task config

### PR DESCRIPTION
**Pull request by cjpatton**
Based on #128 (merge that first).

Add the query type to the task configuration. For fixed-size queries it is necessary to handle batch sizes that exceed the maximum batch size, so do this as well.

While at it, change min_batch_duration to time_precision to match DAP-02.

### CODE REVIEW

File: daphne/src/lib.rs [hunk 1]
```diff
@@ -304,18 +304,80 @@ pub struct DapGlobalConfig {
...
+    /// roughly the same size.
+    FixedSize {
+        min_batch_size: u64,
+        max_batch_size: u64,
+    },
+}
+
+impl DapQueryConfig {
+    /// Check if the batch is ready to aggregate based on the report count. Returns an error if the
+    /// batch size is too large.
```
Comment (chris-wood) [inline daphne/src/lib.rs:327]: Isn't the error returned if the _report count_ is too large (`report_count > *max_batch_size`)?

File: daphne/src/roles.rs [hunk 2]
```diff
@@ -424,8 +424,13 @@ pub trait DapLeader<'a, S>: DapAuthorizedSender<S> + DapAggregator<'a, S> {
             .get_agg_share(&collect_req.task_id, &collect_req.query)
             .await?;
 
-        // Check that the minimum batch size is met.
-        if leader_agg_share.report_count < task_config.min_batch_size {
+        // Check the batch size. If not not ready, then return early.
+        //
+        // TODO Consider logging this error, as it shouuld never happen.
```
Comment (chris-wood) [inline daphne/src/roles.rs:429]: 👍

**Review by chris-wood — APPROVED** [review]
I was a bit confused at first since I expected the new query types to be supported, but this just moves the query config around. 

Pinning a reminder for myself to ask if we have tests for the `InvalidBatchSIze` error for chunky queries. I assume we'll get to that at some point later?

File: daphne/src/lib.rs (hunk 1 above)
Comment (cjpatton) [reply daphne/src/lib.rs:327]: Yup! Done.

**Comment by cjpatton** [comment]
> Pinning a reminder for myself to ask if we have tests for the InvalidBatchSIze error for chunky queries. I assume we'll get to that at some point later?

+1, I'm not sure we do. This should come up in #133.

**Pull request merged**